pkg/api/labels.go

```diff
@@ -55,6 +55,8 @@ const (
 	VersionLabel = "com.docker.compose.version"
 	// ImageBuilderLabel stores the builder (classic or BuildKit) used to produce the image.
 	ImageBuilderLabel = "com.docker.compose.image.builder"
+	// ContainerReplaceLabel is set when container is created to replace another container (recreated)
+	ContainerReplaceLabel = "com.docker.compose.replace"
```

Comment:
Ah this is clever and I like that it lets us reliably associate it with the old one by ID!